src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,584 @@
+     * the specified security provider.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param provider
+     *     the provider to use for this key derivation; if null, this method is
+     *     equivalent to {@code getInstance(String)}
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if a provider is specified and it does not support the specified KDF
+     *     algorithm, or if provider is {@code null} and there is no provider
+     *     that supports a KDF implementation of the specified algorithm
+     * @throws NullPointerException
+     *     if the algorithm is {@code null}
+     */
+    public static KDF getInstance(String algorithm, Provider provider)
+        throws NoSuchAlgorithmException {
+        try {
+            return getInstance(algorithm, null,
+                               provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "Received an InvalidAlgorithmParameterException. Does this "
+                + "algorithm require an AlgorithmParameterSpec?", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm and
+     * is initialized with the specified parameters.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use
+     * @param algParameterSpec
+     *     the {@code AlgorithmParameterSpec} used to configure this KDF's
+     *     algorithm or {@code null} if no additional parameters are provided
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for
+     *     the specified algorithm
+     * @throws InvalidAlgorithmParameterException
+     *     if the {@code AlgorithmParameterSpec} is an invalid value
+     * @throws NullPointerException
+     *     if the algorithm is {@code null}
+     */
+    public static KDF getInstance(String algorithm,
+                                  AlgorithmParameterSpec algParameterSpec)
+        throws NoSuchAlgorithmException, InvalidAlgorithmParameterException {
+        Objects.requireNonNull(algorithm, "null algorithm name");
+        // make sure there is at least one service from a signed provider
+        Iterator<Service> t = GetInstance.getServices("KDF", algorithm);
+        while (t.hasNext()) {
+            Service s = t.next();
+            if (!JceSecurity.canUseProvider(s.getProvider())) {
+                continue;
+            }
+            return new KDF(s, t, algorithm, algParameterSpec);
+        }
+        throw new NoSuchAlgorithmException
+                  ("Algorithm " + algorithm + " not available");
+    }
+
+    /**
+     * Creates an instance of the {@code KDF} object with a specific
+     * {@code Provider}.
```

Comment:
Suggest rewording like other getInstance methods: "Returns a {@code KDF} object that implements the specified algorithm from the specified provider and is initialized with the specified parameters."